src/transports/dash/segment_parser.ts

```diff
@@ -69,18 +76,43 @@ export default function parser({ content,
                                                             0),
                           appendWindow: [period.start, period.end] });
   } else { // it is an initialization segment
+    const { privateInfos } = segment;
+    const shouldExtractCompleteInitChunk = privateInfos !== undefined &&
+                                           privateInfos.shouldGuessInitRange === true;
+
+    const completeInitChunk = shouldExtractCompleteInitChunk ?
+      extractCompleteInitChunk(chunkData) : chunkData;
+
+    if (completeInitChunk === null &&
+        (
+          nextSegments === null ||
+          nextSegments.length === 0
+        )
+    ) {
+      if (!shouldExtractCompleteInitChunk) {
+        throw new Error("Can't extract complete init chunk and segment" +
+                        "references from loaded data.");
```

Comment:
This error is wrong.
You say we could not, except we didn't (because `shouldExtractCompleteInitChunk` was false)

Reply:
Also what's the problem with a `null` initialization segment? Why would we throw here?

Reply:
Done